File: OpenCV/19_estereo/python/main.py
import cv2, numpy as np
L=cv2.imread("left.png",0); R=cv2.imread("right.png",0); assert L is not None and R is not None
st=cv2.StereoSGBM_create(minDisparity=0,numDisparities=16*8,blockSize=5,P1=8*5*5,P2=32*5*5,uniquenessRatio=10,speckleWindowSize=100,speckleRange=2); disp=st.compute(L,R).astype(np.float32)/16.0; np.save("disparity.npy",disp); print("disparidad válida",np.count_nonzero(disp>0))

File: OpenCV/19_estereo/cpp/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
int main(){ auto L=cv::imread("left.png",0),R=cv::imread("right.png",0); if(L.empty()||R.empty()) return 1; auto s=cv::StereoSGBM::create(0,16*8,5,8*25,32*25,1,63,10,100,2,cv::StereoSGBM::MODE_SGBM_3WAY); cv::Mat d16,d; s->compute(L,R,d16); d16.convertTo(d,CV_32F,1.0/16.0); std::cout<<"disp "<<d.size()<<"\n"; }
